ax/modelbridge/transforms/unit_x.py

```diff
@@ -109,102 +110,118 @@ def untransform_observation_features(
+        distribution.distribution_parameters["loc"] = loc
+        distribution.distribution_parameters["scale"] = (
+            distribution.distribution_parameters.get("scale", 1.0)
+            / p_range
+            * self.target_range
+        )
+        # Check that the distribution is valid after the transform.
+        try:
+            distribution.distribution
+        except TypeError:
+            raise UnsupportedError(
+                f"The distribution {str(distribution)} does not support transforming "
+                "via `loc` and `scale` arguments. Consider manually normalizing the "
+                "parameter and the corresponding distribution."
             )
-            # Check that the distribution is valid after the transform.
-            try:
-                dist.distribution
-            except TypeError:
-                raise UnsupportedError(
-                    f"The distribution {str(dist)} does not support transforming via "
-                    "`loc` and `scale` arguments. Consider manually normalizing the "
-                    "parameter and the corresponding distribution."
+
+    def _transform_multivariate_distribution(
+        self, distribution: ParameterDistribution, is_environmental: bool
+    ) -> None:
+        r"""Transform a multivariate distribution in-place."""
+        # Ignore if the ranges of all parameters are same as the target range.
+        if (
+            all(
+                self.bounds[p_name][1] - self.bounds[p_name][0] == self.target_range
+                for p_name in distribution.parameters
+            )
+            and not is_environmental
+        ):
+            return
+        if distribution.distribution_class == "multivariate_normal":
+            # If S is cov and A is the diagonal scale matrix,
+            # the new cov will be ASA.
+            n_dist_params = len(distribution.parameters)
+            scale_vec = np.zeros(len(distribution.parameters))
+            for i, p in enumerate(distribution.parameters):
+                bounds = self.bounds[p]
+                p_range = bounds[1] - bounds[0]
+                scale_vec[i] = self.target_range / p_range
+            cov = np.asarray(distribution.distribution_parameters.get("cov"))
+            if cov.shape != (n_dist_params, n_dist_params):
+                raise UserInputError(
+                    "Expected `cov` to be a square matrix of size equal to "
+                    "number of parameters. Received `cov` with shape "
+                    f"{cov.shape} for the distribution of {n_dist_params} "
+                    "parameters."
                 )
+            distribution.distribution_parameters["cov"] = (
+                scale_vec[..., None] * cov * scale_vec[..., None, :]
+            )
+            mean = np.asarray(distribution.distribution_parameters.get("mean", 0.0))
+            if not np.all(mean == 0):
+                if mean.shape != (n_dist_params,):
```

Comment:
Do we not want to throw the shape error if mean is not of the right shape? Or do we also want to allow passing scalars? I think we can be pretty restrictive in what we support here as it's deep down in the code (if we wanted to support user friendly interfaces we could do that further up in the stack).

Reply:
I'd generally expect the non-zero mean case to be limited to environmental variables only. Since this is an advanced feature anyway, I prefer to be restrictive and require the user to be explicit in what they want to achieve.